src/cloudabi-run/cloudabi-run.cc

```diff
@@ -79,6 +81,15 @@ int main(int argc, char *argv[]) {
       &argdata_factory);
   YAMLBuilder<const argdata_t *> builder(&canonicalizing_factory);
   const argdata_t *ad;
+
+  execfd = open(argv[0], O_EXEC);
+  if (execfd == -1) {
+    std::cerr << "Failed to open executable: " << std::strerror(errno)
+              << std::endl;
+    return 127;
+  }
+  file_descriptor_factory.SetExecutableFd(execfd);
```

Comment:
Could you move this block of code further up, so that the file descriptor can be passed to the constructor of `YAMLFileDescriptorFactory`?